pulp_smash/api.py

```diff
@@ -265,6 +265,41 @@ def task_handler(client, response):
     return done_task
 
 
+def smart_handler(client, response):
+    """Decides which handler to call based on response content.
+
+    Do the following:
+
+    1. Pass response through safe_handler to handle 202 and raise_for_status.
+    2. Return the response if it is not Pulp 3.
+    3. Return the response if it is not application/json type.
+    4. Pass response through task_handler if is JSON 202 with 'task'.
+    5. Pass response through page_handler if is JSON but not 202 with 'task'.
+    """
+    # safe_handler Will raise_for_Status, handle 202 and pool tasks
+    response = safe_handler(client, response)
+
+    try:
+        check_pulp3_restriction(client)
+    except ValueError:
+        # If pulp is not 3+ return the result of safe_handler by default
+        return response
+
+    ctype = response.headers.get("Content-Type")
+    is_202 = response.status_code == 202
+
+    if ctype != "application/json":
+        # Not a valid JSON, return pure response
+        return response
+
+    # We got JSON is that a task call report?
```

Comment:
Move this `is_202` closer.
```suggestion
    # We got JSON is that a task call report?
    is_202 = response.status_code == 202
```

is_202 = response.status_code == 202